src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -144,6 +152,8 @@ void Leader::HandleServerData(Coap::Message &aMessage, const Ip6::MessageInfo &a
     ThreadNetworkDataTlv networkDataTlv;
     uint16_t             rloc16;
 
+    VerifyOrExit(!mWaitingForNetDataSync);
```

Comment:
Put this after the "received network data registration" log line below? I think it may be useful to know that the message was received but then rejected.